pkg/batch/jobs/opsevent/datetime_watcher.go

```diff
@@ -49,45 +51,78 @@ func NewDatetimeWatcher(
 		opt(dopts)
 	}
 	return &datetimeWatcher{
-		environmentLister: targetStore,
-		autoOpsRuleLister: targetStore,
-		autoOpsExecutor:   autoOpsExecutor,
-		opts:              dopts,
-		logger:            dopts.Logger.Named("datetime-watcher"),
+		envClient:       envClient,
+		aoClient:        aoClient,
+		autoOpsExecutor: autoOpsExecutor,
+		opts:            dopts,
+		logger:          dopts.Logger.Named("datetime-watcher"),
 	}
 }
 
 func (w *datetimeWatcher) Run(ctx context.Context) (lastErr error) {
 	ctx, cancel := context.WithTimeout(ctx, w.opts.Timeout)
 	defer cancel()
-	environments := w.environmentLister.GetEnvironments(ctx)
-	for _, env := range environments {
-		autoOpsRules := w.autoOpsRuleLister.GetAutoOpsRules(ctx, env.Namespace)
+	envs, err := w.listEnvironments(ctx)
+	if err != nil {
+		lastErr = err
+		return
+	}
+	for _, env := range envs {
+		autoOpsRules, err := w.listAutoOpsRules(ctx, env.Id)
+		if err != nil {
+			lastErr = err
+			return
+		}
 		for _, a := range autoOpsRules {
-			asmt, err := w.assessAutoOpsRule(ctx, env, a)
+			aor := &autoopsdomain.AutoOpsRule{AutoOpsRule: a}
+			asmt, err := w.assessAutoOpsRule(ctx, env.Id, aor)
 			if err != nil {
 				lastErr = err
 			}
 			if !asmt {
 				continue
 			}
-			if err = w.autoOpsExecutor.Execute(ctx, env.Namespace, a.Id); err != nil {
+			if err = w.autoOpsExecutor.Execute(ctx, env.Id, a.Id); err != nil {
 				lastErr = err
 			}
 		}
 	}
 	return
 }
 
+func (w *datetimeWatcher) listEnvironments(ctx context.Context) ([]*envproto.EnvironmentV2, error) {
+	resp, err := w.envClient.ListEnvironmentsV2(ctx, &envproto.ListEnvironmentsV2Request{
+		PageSize: 0,
```

Comment:
To make it behave like the V1 API, the following line needs to be added.
```suggestion
		PageSize: 0,
		Archived:  wrapperspb.Bool(false)
```

The V1 API always filtered out the deleted environments.
Since the V2 environment has the `archived` field instead of the `deleted` field, we need to specify where including or not `archived` environments.

Reply:
I have added it.
Thanks!